rpmlint/checks/MenuXDGCheck.py

```diff
@@ -17,51 +17,33 @@
 
 
 class MenuXDGCheck(AbstractFilesCheck):
+    """
+    Check whether MenuXDG files installed by a package are valid.
+    """
     def __init__(self, config, output):
         # desktop file need to be in $XDG_DATA_DIRS
         # $ echo $XDG_DATA_DIRS/applications
         # /var/lib/menu-xdg:/usr/share
-        super().__init__(config, output, r'(?:/usr|/etc/opt|/opt/.*)/share/applications/.*\.desktop$')
+        super().__init__(config, output, r'(?:/usr/share)/applications/.*\.desktop$')
 
     def parse_desktop_file(self, pkg, root, f, filename):
+        """
+        Check the structure of a desktop file.
+        """
         cfp = cfgparser.RawConfigParser()
         try:
             with codecs.open(f, encoding='utf-8') as inputf:
                 cfp.read_file(inputf, filename)
-        except cfgparser.DuplicateSectionError as e:
-            self.output.add_info('E',
-                                 pkg, 'desktopfile-duplicate-section', filename,
-                                 '[%s]' % e.section)
-        except cfgparser.MissingSectionHeaderError:
-            self.output.add_info('E', pkg, 'desktopfile-missing-header', filename)
         except cfgparser.Error as e:
-            # Only in Python >= 3.2
-            if (hasattr(cfgparser, 'DuplicateOptionError') and
-                    isinstance(e, cfgparser.DuplicateOptionError)):
-                self.output.add_info('E', pkg, 'desktopfile-duplicate-option', filename,
-                                     '[%s]/%s' % (e.section, e.option))
-            else:
-                self.output.add_info('W', pkg, 'invalid-desktopfile', filename,
-                                     e.message.partition(':')[0])
+            self._handle_parser_error(pkg, filename, e)
         except UnicodeDecodeError as e:
+            if not is_utf8(f):  # not optimal, but working
+                self.output.add_info('E', pkg, 'non-utf8-desktopfile', filename)
             self.output.add_info('W', pkg, 'invalid-desktopfile', filename, 'Unicode error: %s' % (e))
```

Comment:
Why double check, if one tool reports unicod decoding then just throw out unicode parsing error and be done with it.